### DISCUSSION TITLE
Search in specific files onedrive

### DISCUSSION THREAD

**Discussion by Patodif** [microsoft-search/pnp-modern-search#3281] [Q&A]
Hello, I am trying to make a search engine but only for a certain OneDrive folder, when I use the expression {searchTerms} path:https://xxxxxxx-my.sharepoint.com/:f:/g/xxxxxxxxx/xxxxxxxxxxxx/Epd8xhwSS3dNmcYb1KIIJJkBrbNcfqhSnp2dMmgDq2055g?e =DVfhMN, I don't get any results. Only by using {searchTerms} I can see all the files. I need to restrict the search to a local file and a shared file.

**Comment by kasperbolarsen** [comment]
Hi,
The URL you get from sharing links and the like can't be used in Search. In Search you must have the proper URL and one way to find it is to open the folder in question, click the 3 dots next to a document, select "Details", scroll the right hard side panel to the "Path" section:
Remove the document related part from the URL and test the remaining URL in the SP Search Query tool in order to test that you get the expected result :-)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/5ef05e92-6266-4236-b795-0cd8026d222a)

**Comment by Patodif** [comment]
Thank you very much...if I share the file I can search for it, but if it has been shared with me it does not allow me to search for it, although it does when I do not restrict the search...is this correct?
Can I request help again? Why don't you look for videos on how I can do it?
Again, thank you so much
________________________________
De: Kasper Larsen ***@***.***>
Enviado: domingo, 1 de octubre de 2023 15:29
Para: microsoft-search/pnp-modern-search ***@***.***>
Cc: Patodif ***@***.***>; Manual ***@***.***>
Asunto: Re: [microsoft-search/pnp-modern-search] Search in specific files onedrive (Discussion #3281)


Hi,
The URL you get from sharing links and the like can't be used in Search. In Search you must have the proper URL and one way to find it is to open the folder in question, click the 3 dots next to a document, select "Details", scroll the right hard side panel to the "Path" section:
Remove the document related part from the URL and test the remaining URL in the SP Search Query tool in order to test that you get the expected result :-)
[image]<https://user-images.githubusercontent.com/20593570/271837781-5ef05e92-6266-4236-b795-0cd8026d222a.png>

**Reply by kasperbolarsen** [reply]
I haven't worked a lot with OneDrive and shared files in a Search context,  but perhaps using the Microsoft Search data source would be an idea in your case. Microsoft search is based on Graph and this doc indicates that Graph might contain the data you are looking for: https://learn.microsoft.com/en-us/graph/api/drive-sharedwithme?view=graph-rest-1.0&tabs=http

**Comment by Patodif** [comment]
Thank you very much again, I will read the article. The last question is possible...hahaha.
Now that I can apply the filters, the results always appear in the search engine with the search bar empty. Do you know how I can leave it blank?
Thanks....
[cid:b03da22f-4ee7-4826-95f1-6c0f947cfa54]
________________________________

**Reply by kasperbolarsen** [reply]
Hard to offer solid advise without knowing your setup. If your results web part, pane 3 looks like this you should disable the "Use a default value"

![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/5f276a16-a8e1-4b00-8d8a-acd487ef80f3)

**Comment by Patodif** [comment]
Thanks a lot, i will try with this. How can i see the path for a sharepoint list?, I want make the search only in a specific list, is that possible?
Thanks
Bye
________________________________
De: Kasper Larsen ***@***.***>
Enviado: domingo, 8 de octubre de 2023 15:14
Para: microsoft-search/pnp-modern-search ***@***.***>
Cc: Patodif ***@***.***>; Manual ***@***.***>
Asunto: Re: [microsoft-search/pnp-modern-search] Search in specific files onedrive (Discussion #3281)


Hard to offer solid advise without knowing your setup. If your results web part, pane 3 looks like this you should disable the "Use a default value"

[image]<https://user-images.githubusercontent.com/20593570/273457406-5f276a16-a8e1-4b00-8d8a-acd487ef80f3.png>

**Reply by kasperbolarsen** [reply]
The URL of the List  is visible in the browser when visiting the list

**Comment by Patodif** [comment]
https://XXXXXXXXXXXX.sharepoint.com/sites/XXXXXXXXX/Lists/Listado%20telefonos/AllItems.aspx
If I put only this i haven´t 0 results or is there a spacial path?

________________________________
De: Kasper Larsen ***@***.***>
Enviado: domingo, 8 de octubre de 2023 16:49
Para: microsoft-search/pnp-modern-search ***@***.***>
Cc: Patodif ***@***.***>; Manual ***@***.***>
Asunto: Re: [microsoft-search/pnp-modern-search] Search in specific files onedrive (Discussion #3281)


The URL of the List is visible in the browser when visiting the list

**Reply by kasperbolarsen** [reply]
Try path:https://xxxxxxxxxxxx.sharepoint.com/sites/XXXXXXXXX/Lists/Listado%20telefonos*

**Comment by Patodif** [comment]
You are a genious

________________________________
De: Kasper Larsen ***@***.***>
Enviado: domingo, 8 de octubre de 2023 17:01
Para: microsoft-search/pnp-modern-search ***@***.***>
Cc: Patodif ***@***.***>; Manual ***@***.***>
Asunto: Re: [microsoft-search/pnp-modern-search] Search in specific files onedrive (Discussion #3281)


Try path:https://xxxxxxxxxxxx.sharepoint.com/sites/XXXXXXXXX/Lists/Listado%20telefonos*

**Comment by Patodif** [comment]
Still i see the results without search make...i look the information in 3 panel and nothig work yet...Now i try to see on column the list but i can change that yet

________________________________
De: Patricia Di Francesco ***@***.***>
Enviado: domingo, 8 de octubre de 2023 17:06
Para: microsoft-search/pnp-modern-search ***@***.***>
Asunto: RE: [microsoft-search/pnp-modern-search] Search in specific files onedrive (Discussion #3281)


You are a genious

________________________________
De: Kasper Larsen ***@***.***>
Enviado: domingo, 8 de octubre de 2023 17:01
Para: microsoft-search/pnp-modern-search ***@***.***>
Cc: Patodif ***@***.***>; Manual ***@***.***>
Asunto: Re: [microsoft-search/pnp-modern-search] Search in specific files onedrive (Discussion #3281)


Try path:https://xxxxxxxxxxxx.sharepoint.com/sites/XXXXXXXXX/Lists/Listado%20telefonos*

**Comment by Patodif** [comment]
[cid:a33a1c4b-883f-42f5-965d-e0cb468ac640]
This is empty and have results can this wil be empty?
[cid:ecd4521f-d440-444e-9112-3273453c5b2c]
This the origin list and if i look the list only see the "departamento " column, how can i change that?
Thanks
________________________________
De: Patricia Di Francesco ***@***.***>
Enviado: domingo, 8 de octubre de 2023 17:06
Para: microsoft-search/pnp-modern-search ***@***.***>
Asunto: RE: [microsoft-search/pnp-modern-search] Search in specific files onedrive (Discussion #3281)


You are a genious

________________________________
De: Kasper Larsen ***@***.***>
Enviado: domingo, 8 de octubre de 2023 17:01
Para: microsoft-search/pnp-modern-search ***@***.***>
Cc: Patodif ***@***.***>; Manual ***@***.***>
Asunto: Re: [microsoft-search/pnp-modern-search] Search in specific files onedrive (Discussion #3281)


Try path:https://xxxxxxxxxxxx.sharepoint.com/sites/XXXXXXXXX/Lists/Listado%20telefonos*

**Reply by kasperbolarsen** [reply]
? Should there have been any screen shots?

**Comment by Patodif** [comment]
This is the list
[cid:1000fdbe-af73-49ef-9237-404abdc31f4e]
And the PNP Results only shows the columns DEPARTAMENTO
[cid:82abc48c-96a9-4817-89ad-507c7759195d]
________________________________
De: Kasper Larsen ***@***.***>
Enviado: domingo, 8 de octubre de 2023 18:08
Para: microsoft-search/pnp-modern-search ***@***.***>
Cc: Patodif ***@***.***>; Manual ***@***.***>
Asunto: Re: [microsoft-search/pnp-modern-search] Search in specific files onedrive (Discussion #3281)


? Should there have been any screen shots?

**Reply by kasperbolarsen** [reply]
Please verify using one of the tools from this guide, https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/

**Comment by Patodif** [comment]
Maybe the mistake is in the list not in the results?...i will try again
________________________________
De: Kasper Larsen ***@***.***>
Enviado: domingo, 8 de octubre de 2023 18:26
Para: microsoft-search/pnp-modern-search ***@***.***>
Cc: Patodif ***@***.***>; Manual ***@***.***>
Asunto: Re: [microsoft-search/pnp-modern-search] Search in specific files onedrive (Discussion #3281)


Please verify using one of the tools from this guide, https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/

**Comment by Patodif** [comment]
Hi, one questions, how can i do to hidden one lista or file to PnP Search?
Thanks
Bye

Enviado desde Outlook para Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by kasperbolarsen** [reply]
This returns all content from this site collection: 
path:https://tcwlv.sharepoint.com/sites/PnPModernSearch*

This returns all content except the content in the LibraryWithCheckout library:
path:https://tcwlv.sharepoint.com/sites/PnPModernSearch* -path:https://tcwlv.sharepoint.com/sites/PnPModernSearch/LibraryWithCheckout/*



This returns all content from the site collection except the file "AB12-2.pdf"
path:https://tcwlv.sharepoint.com/sites/PnPModernSearch* -path:https://tcwlv.sharepoint.com/sites/PnPModernSearch/DocsWithContentType/AB12-2.pdf
